### DISCUSSION TITLE
Conditionally convert if alpha is empty

### DISCUSSION THREAD

**Discussion by TomQv** [ImageMagick/ImageMagick#4456] [Digital Image Processing]
Hi Guys, is there a way to conditionally convert RGBA-pngs to RGB-pngs only for those pngs, where the alpha channel is empty?

**Comment by fmw42** [comment]
What version of ImageMagick and what platform/OS?

You can test the alpha channel to see if fully opaque and then convert those.

For IM 6 unix syntax,

```
test=$(convert image.png -channel a -separate -format "%[fx:mean==1?1:0]" info:)
if [ $test -eq 1 ]; then
convert image.png -alpha off image.png
fi

```

**Comment by TomQv** [comment]
Thanks for quick answer, this looks good.
I'm a complete beginner to IM,  will try to find out how this goes in Windows 10
